### DISCUSSION TITLE
Peer Discovery issues using 3.13.x in MS Azure Service Fabric

### DISCUSSION THREAD

**Discussion by abeckham7305** [rabbitmq/rabbitmq-server#11653] [Questions] [answered]
I'm running an application in Microsoft Azure Service Fabric with three nodes, using RabbitMQ as my message broker.  I have the nodes in rabbitmq.conf configured using the simple classic_config option.  I specify the three nodes there as follows:

cluster_formation.peer_discovery_backend = rabbit_peer_discovery_classic_config
cluster_formation.classic_config.nodes.1 = rabbit@node1
cluster_formation.classic_config.nodes.2 = rabbit@node2
cluster_formation.classic_config.nodes.3 = rabbit@node3
cluster_formation.node_type = disc

This worked fine for me in 3.12.x versions of RabbitMQ.  The nodes would join the same cluster whether I was upgrading the RabbitMQ version or installing it new.

In version 3.13.x this doesn't work for me anymore when installing new.  If upgrading it works fine since the nodes have already been discovered and stored in the data files.  I believe the issue is with recent updates to how peer discovery works.  I don't understand the code enough to figure this out myself.  It seems as if the new peer discovery method is starting a hidden node to perform the clustering and this hidden node is using the default erlang cookie for the user instead of the one I am specifying as a parameter in Service Fabric to pass to RabbitMQ.  I'm passing this cookie value using this method for my parameter:

RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-setcookie abcd-efgh-1234-5678-etc"

The errors I keep getting are like this, which I read are typically related to mismatched erlang cookies:
** Connection attempt from node 'rabbit-930-4988@node2' rejected. Invalid challenge reply. **
** Connection attempt from node 'rabbit-706-6372@node3' rejected. Invalid challenge reply. **

Any thoughts on whether this is a bug or if I should be doing something differently?

**Comment by lukebakken** [comment, marked as answer]
Thank you for your comprehensive report. Indeed, the new peer discovery code does not check `RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS` for `-setcookie`, but it _DOES_ use [`RABBITMQ_ERLANG_COOKIE`](https://github.com/rabbitmq/rabbitmq-server/blob/6b6cf58af842e98cd36918efa7998742617ea107/deps/rabbit_common/src/rabbit_env.erl#L1488-L1501) in the [peer discovery code](https://github.com/rabbitmq/rabbitmq-server/blob/01d248f2d1235fbe0bfdee89e7615d2c019aeca9/deps/rabbit/src/rabbit_peer_discovery.erl#L385-L391).

So, I'm pretty sure your issue will be resolved if you remove `-setcookie` from `RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS` (or delete that env var if it's only used for the cookie), and, instead, use `RABBITMQ_ERLANG_COOKIE`.

Please let me know how this works.

**Reply by lukebakken** [reply]
PR to allow using `RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS`: https://github.com/rabbitmq/rabbitmq-server/pull/11654

**Reply by abeckham7305** [reply]
Thank you for the quick reply!  I'll test your recommendation out today and let you know the results.

**Reply by abeckham7305** [reply]
Using RABBITMQ_ERLANG_COOKIE instead totally worked!  Thanks again!!

**Reply by lukebakken** [reply]
Thank you for the follow-up.
